python/cudf/cudf/_libxx/sort.pyx

```diff
@@ -195,3 +198,73 @@ def digitize(Table source_values_table, Table bins, bool right=False):
             )
 
     return Column.from_unique_ptr(move(c_result))
+
+
+def rank_columns(Table source_table, str method, str na_option, bool ascending,
+                 bool pct
+                 ):
+    """
+    Compute numerical data ranks (1 through n) of each column in the dataframe
+    """
+    cdef table_view source_table_view = source_table.data_view()
+
+    cdef rank_method c_rank_method
+    if method == 'min':
+        c_rank_method = rank_method.MIN
+    elif method == 'max':
+        c_rank_method = rank_method.MAX
+    elif method == 'first':
+        c_rank_method = rank_method.FIRST
+    elif method == 'dense':
+        c_rank_method = rank_method.DENSE
+    else:
+        c_rank_method = rank_method.AVERAGE
+
+    cdef order column_order = (
+        order.ASCENDING
+        if ascending
+        else order.DESCENDING
+    )
+    # ascending
+    #    #top    = na_is_smallest
+    #    #bottom = na_is_largest
+    #    #keep   = na_is_largest
+    # descending
+    #    #top    = na_is_largest
+    #    #bottom = na_is_smallest
+    #    #keep   = na_is_smallest
+    cdef null_order null_precedence
+    if ascending:
+        if na_option == 'top':
+            null_precedence = null_order.BEFORE
+        else:
+            null_precedence = null_order.AFTER
+    else:
+        if na_option == 'top':
+            null_precedence = null_order.AFTER
+        else:
+            null_precedence = null_order.BEFORE
+    cdef include_nulls _include_nulls = (
+        include_nulls.EXCLUDE_NULLS
+        if na_option == 'keep'
+        else include_nulls.INCLUDE_NULLS
+    )
+    cdef unique_ptr[table] c_result
+    cdef bool percentage = True if pct else False
+
+    with nogil:
+        c_result = move(
+            rank(
+                source_table_view,
+                c_rank_method,
+                column_order,
+                _include_nulls,
+                null_precedence,
+                percentage
+            )
+        )
+
+    return Table.from_unique_ptr(
+        move(c_result),
+        column_names=source_table._column_names
+    )
```

Comment:
We're losing the index here. We need to attach it to this table via something like:
```suggestion
    out_table = Table.from_unique_ptr(
        move(c_result),
        column_names=source_table._column_names
    )
    out_table._index = source_table._index
    return out_table
```